crates/cxx-qt-gen/src/syntax/implitemmethod.rs

```diff
@@ -4,16 +4,16 @@
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
 use syn::{
-    FnArg, GenericArgument, ImplItemMethod, Pat, PatIdent, PatType, PathArguments, Receiver, Type,
+    FnArg, GenericArgument, Pat, PatIdent, PatType, PathArguments, Receiver, Signature, Type,
     TypePath, TypeReference,
 };
 
 /// Return whether the first parameter of a method is a mutable self argument
 //
 // Note that self: Box<Self> is parsed as FnArg::Typed not FnArg::Receiver so will be false
 // but we don't use this case with CXX, so this can be ignored.
-pub fn is_method_mutable(method: &ImplItemMethod) -> bool {
-    match method.sig.inputs.first() {
+pub fn is_method_mutable(signature: &Signature) -> bool {
```

Comment:
also feels like all of this could be separated out as well if we are separating things :-)

Reply:
Done